rust/threshold-schnorr/src/schnorr_example_rust/src/wasm_only.rs

```diff
@@ -118,6 +186,43 @@ fn verify_bip340_secp256k1(
     Ok(SignatureVerificationReply { is_signature_valid })
 }
 
+fn verify_bip341_secp256k1(
+    sig_bytes: &[u8],
+    msg_bytes: &[u8],
+    secp1_pk_bytes: &[u8],
+    merkle_tree_root_bytes: &[u8],
+) -> Result<SignatureVerificationReply, String> {
+    assert_eq!(secp1_pk_bytes.len(), 33);
+
+    let pk = XOnlyPublicKey::from_slice(&secp1_pk_bytes[1..]).unwrap();
+    let tweaked_pk_bytes = {
+        let secp256k1_engine = Secp256k1::new();
+        let merkle_root = if merkle_tree_root_bytes.len() == 0 {
+            None
+        } else {
+            Some(
+                bitcoin::hashes::Hash::from_slice(&merkle_tree_root_bytes)
+                    .expect("failed to create TapBranchHash"),
+            )
+        };
+
+        pk.tap_tweak(&secp256k1_engine, merkle_root)
+            .0
+            .to_inner()
+            .serialize()
+    };
+
+    let sig =
+        k256::schnorr::Signature::try_from(sig_bytes).expect("failed to deserialize signature");
+
+    let vk = k256::schnorr::VerifyingKey::from_bytes(&tweaked_pk_bytes)
+        .expect("failed to deserialize BIP340 encoding into public key");
```

Comment:
IIRC we can do the entire verification using `secp256k1` here rather than having to use both `k256` and `secp256k`.

Reply:
Alternately, with a bit more effort (copying the code from https://github.com/dfinity/ic/blob/master/rs/crypto/secp256k1/src/lib.rs#L352) we could do the verification using only `k256` and avoid the `bitcoin` dependency. Given the apparent complications in the build that this brings in, it might be worthwhile.